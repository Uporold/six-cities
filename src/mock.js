// eslint-disable-next-line import/prefer-default-export
export const fakeAdTitles = [
  `Beautiful & luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];
