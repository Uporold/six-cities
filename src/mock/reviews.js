export const reviews = [
  {
    id: 1,
    user: {
      id: 14,
      isPro: true,
      name: "Corey",
      avatar:
        "https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg",
    },
    rating: 4,
    comment:
      "Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius",
    date: "2020-07-29T07:00:10.725Z",
  },
  {
    id: 2,
    user: {
      id: 12,
      isPro: true,
      name: "Isaac",
      avatar:
        "https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/3.jpg",
    },
    rating: 4,
    comment:
      "The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.",
    date: "2020-07-29T07:00:10.725Z",
  },
];
