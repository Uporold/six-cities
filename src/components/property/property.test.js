import React from "react";
import renderer from "react-test-renderer";
import Property from "./property";
import { hotels } from "../../mock/offers";

it(`Should Property Page render correctly`, () => {
  const tree = renderer
    .create(
      <Property hotel={hotels[0]} hotels={hotels} onPlaceCardClick={() => {}} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
