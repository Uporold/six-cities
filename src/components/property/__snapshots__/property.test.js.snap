// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Property Page render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  test@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Nice, cozy, warm big bed apartment
            </h1>
            <button
              className="property__bookmark-button  button"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "98%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.9
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              Entire place
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              6
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              488
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Breakfast
              </li>
              <li
                className="property__inside-item"
              >
                Baby seat
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Fridge
              </li>
              <li
                className="property__inside-item"
              >
                Air conditioning
              </li>
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
              <span
                className="property__user-status"
              >
                Pro
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews ·
              <span
                className="reviews__amount"
              >
                2
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper  property__avatar-wrapper--pro"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Corey
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    Home is amazing. It's like staying in a museum. The rooms, furnishings and artworks are incredible. The views of My Vesuvius
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-07-29T07:00:10.725Z"
                  >
                    July 2020
                  </time>
                </div>
              </li>
              <li
                className="reviews__item"
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper  property__avatar-wrapper--pro"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/3.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Isaac
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "80%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime="2020-07-29T07:00:10.725Z"
                  >
                    July 2020
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form"
              method="post"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
                onChange={[Function]}
              >
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value={5}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value={4}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value={3}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value={2}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  disabled={false}
                  id="1-stars"
                  name="rating"
                  type="radio"
                  value={1}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set
                   
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
      >
        <div
          style={
            Object {
              "height": "100%",
            }
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseOut={[Function]}
          >
            
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offers/2"
                onClick={[Function]}
              >
                <img
                  alt="Penthouse, 4-5 rooms + 5 balconies"
                  className="place-card__image"
                  height="200"
                  src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className=" place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    327
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "58%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offers/2"
                  onClick={[Function]}
                >
                  Penthouse, 4-5 rooms + 5 balconies
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                House
              </p>
            </div>
          </article>
          <article
            className="near-places__card place-card"
            onMouseEnter={[Function]}
            onMouseOut={[Function]}
          >
            
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offers/3"
                onClick={[Function]}
              >
                <img
                  alt="House in countryside"
                  className="place-card__image"
                  height="200"
                  src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className=" place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    461
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button  button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    To bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "72%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offers/3"
                  onClick={[Function]}
                >
                  House in countryside
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                House
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
